nucliadb/src/nucliadb/ingest/orm/processor/__init__.py

```diff
@@ -182,8 +184,8 @@ async def delete_resource(
                     shard = await kb.get_resource_shard(shard_id)
                     if shard is None:
                         raise AttributeError("Shard not available")
-
-                    await self.shard_manager.delete_resource(
+                    await self._maybe_external_index_delete_resource(message.kbid, uuid)
+                    await self.index_node_shard_manager.delete_resource(
```

Comment:
Shouldn't this operations be exclusive? Why do we need to delete a resource on the nodes if we are using an external index?

Reply:
It's not exclusive because we are still using the texts index (for catalog purposes).
Once we rewrite catalog in a pg index, we will be able to do it exclusive

Reply:
Oh that's true...